test/utils/runners.go

```diff
@@ -52,6 +52,46 @@ const (
 	nonExist = "NonExist"
 )
 
+func WaitUntilPodIsScheduled(c clientset.Interface, name, namespace string, timeout time.Duration) (*v1.Pod, error) {
+	// Wait until it's scheduled
+	p, err := c.Core().Pods(namespace).Get(name, metav1.GetOptions{ResourceVersion: "0"})
+	if err == nil && p.Spec.NodeName != "" {
+		return p, nil
+	}
+	pollingPeriod := 200 * time.Millisecond
```

Comment:
This is only the frequency of checks. I don't see how it can impact flakyness. Timeout is an important thing for that, and it's set for 2 minutes, so waay more than necessary (hopefully).